src/main/java/spoon/reflect/reference/CtTypeParameterReference.java

```diff
@@ -99,4 +99,10 @@
 	@Override
 	@UnsettableProperty
 	<T extends CtActualTypeContainer> T setActualTypeArguments(List<? extends CtTypeReference<?>> actualTypeArguments);
+
+	/**
+	 * Returns true if the given typeReference is the default bounding type (i.e. java.lang.Object)
+	 */
+	@DerivedProperty
+	boolean isDefaultBoundingType(CtTypeReference typeReference);
```

Comment:
parameter needed? could be simply `isDefaultBound()` with no parameter?

Reply:
The parameter is needed for case like https://github.com/INRIA/spoon/pull/1672/files#diff-e14ef777963d781b479e6fdd3e8d419dR79 where we want to check before setting the boundingType.

Reply:
The code on the link will work if object is sent to adapt method too. Only check for null is needed.
The implementation of `is default bounding type` can be made static. May be we do not need such method at all.